### PR TITLE
call VW from R - Useful for computing AUC on validation set and parameter optimization.

### PR DESCRIPTION
**Pull request by martinbel**
This is a simple function to call VW from R, so far for logistic regression. Also it allows to use perf to compute the AUC on a validation set. I've found it a simple way to tune parameters based on a holdout set. I know vw has it's own methods for validation but I believe this is a common task. It's not really a new feature for VW but a simple function to facilitate validating models based on AUC. 
It's also fairly extendable to other problems.
I have written a reproducible example to show how it works. I'll add it as another file.

### CODE REVIEW

**Comment by martinbel** [comment]
Here is a reproducible example of the usage of vw.R. I'm not sure if dt2vw.R is updated to it's newest version, but I've added to a package I'm using (handy).

```
rm(list = ls(all = TRUE)); gc()

library(ggplot2)

# devtools::install_github('martinbel/handy')
library(handy)
# has dt2vw.R from Vw
# ?get_feature_type
# ?handy::vw function in this example.
library(data.table)

dt = diamonds
data_types = handy::get_feature_type(dt, threshold = 50)
namespaces = list(n = list(varName = data_types$num_vars, keepSpace=F),
                  c = list(varName = data_types$fact_vars, keepSpace=F))
dt = setDT(dt)
dt$y = with(dt, ifelse(y < 5.71, 1, -1))
dt2vw(dt, '../diamonds.vw', namespaces, target='y', weight=NULL)
system('head -3 ../diamonds.vw')

# prepare dataset for validation

system('head -10000 ../diamonds.vw > ../X_train.vw ')
system('tail -43940 ../diamonds.vw > ../X_valid.vw ')
write.table(dt$y, file='../valid_labels.txt', row.names = F,
            col.names = F, quote = F)

auc = vw(training_data='../X_train.vw',
         validation_data='../X_valid.vw',
         validation_labels = "../valid_labels.txt",
         loss='logistic', model='mdl.vw', b=25, learning_rate=0.5,
         passes=20, l1=1e-08, l2=1e-08, early_terminate=2, extra='')


### create a parameter grid

grid = expand.grid(l1=c(1e-06, 1e-07, 1e-08),
                   l2=c(1e-06, 1e-07, 1e-08))

aucs <- lapply(1:nrow(grid), function(i){
  g = grid[i, ]
  auc = vw(training_data='../X_train.vw',
           validation_data='../X_valid.vw',
           validation_labels = "../valid_labels.txt",
           loss='logistic', model='mdl.vw', b=25, learning_rate=0.5,
           passes=20, l1=g[['l1']], l2=g[['l2']],
           early_terminate=2, extra='')
  auc
})

cbind(grid, auc=do.call(rbind, aucs))
# l1    l2     auc
# 1 1e-06 1e-06 0.68485
# 2 1e-07 1e-06 0.68169
# 3 1e-08 1e-06 0.68193
# 4 1e-06 1e-07 0.68509
# 5 1e-07 1e-07 0.68185
# 6 1e-08 1e-07 0.68214
# 7 1e-06 1e-08 0.68512
# 8 1e-07 1e-08 0.68187
# 9 1e-08 1e-08 0.68216
```

**Comment by JohnLangford** [comment]
Thanks, I added your example into the repository as well.

-John
